src/steamship/agents/service/agent_service.py

```diff
@@ -295,6 +306,33 @@ def build_default_context(self, context_id: Optional[str] = None, **kwargs) -> A
         context = with_llm(context=context, llm=llm)
         return context
 
+    @post("async_prompt")
+    def async_prompt(
+        self, prompt: Optional[str] = None, context_id: Optional[str] = None, **kwargs
+    ) -> StreamingResponse:
+        with self.build_default_context(context_id, **kwargs) as context:
+            ctx_id = context_id
+
+            # if no context ID is provided, we need to make sure that the streaming context ID
+            # is the same one as the non-streaming.
+            if not ctx_id:
+                ctx_file = context.chat_history.file
+                for tag in ctx_file.tags:
+                    if tag.kind == TagKind.CHAT and tag.name == ChatTag.CONTEXT_KEYS:
+                        if value := tag.value:
+                            ctx_id = value.get("id", None)
+
+            # if you can't find a consistent context_id, then there is no way to provide an accurate
+            # streaming endpoint.
+            if not ctx_id:
+                # TODO(dougreid): this points to a slight flaw in the context_keys vs. context_id
+                raise SteamshipError("Error setting up context: no id found for context.")
+
+            task = self.invoke_later(
+                "/prompt", arguments={"prompt": prompt, "context_id": ctx_id, **kwargs}
+            )
```

Comment:
Forgive me if I'm being dense, but what do we gain out of making this `invoke_later`?

Reply:
We use `invoke_later` to schedule the work and allow an "early" return with the bits required to monitor that work.

 We call `/prompt` but have passed back the file that all the updates (status, and otherwise) will be posted to, but also have a handle on the `Task` that is doing the actual work.

I _thought_ this was the approach discussed earlier. Are you imagining something different?